Sources/Plasma/NucleusLib/pnKeyedObject/plKeyImp.h

```diff
@@ -172,6 +177,9 @@ class plKeyImp : public plKeyData
     mutable int16_t             fPendingRefs;   // Outstanding requests I have out.
     mutable std::vector<plKeyImp*>  fClones;    // clones of me
     mutable plKey               fCloneOwner;    // pointer for clones back to the owning key
+
+    friend class pfConsoleActiveRefPeeker;
+    friend class plPageOptimizer; // needs to update fStartPos
```

Comment:
I don't love these, but they're infinitely better than the UB we were relying on before...  :trollface: